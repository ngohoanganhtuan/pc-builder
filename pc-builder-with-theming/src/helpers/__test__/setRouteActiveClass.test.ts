// Helpers
import { setRouteActiveClass } from '@helpers';

describe('Test [setRouteActiveClass] helper', () => {
  test('Helper [setRouteActiveClass] default class name should match snapshot when active', () => {
    expect(
      setRouteActiveClass({
        isActive: true,
      })
    ).toMatchSnapshot();
  });

  test('Helper [setRouteActiveClass] default class name should match snapshot when not active', () => {
    expect(
      setRouteActiveClass({
        isActive: false,
      })
    ).toMatchSnapshot();
  });

  test('Helper [setRouteActiveClass] should return class names contains "bg-warning border-0" class names when active', () => {
    expect(
      setRouteActiveClass({
        isActive: true,
      })
    ).toContain('bg-warning border-0');
  });

  test('Helper [setRouteActiveClass] should return class names contains "bg-white" class names when not active', () => {
    expect(
      setRouteActiveClass({
        isActive: false,
      })
    ).toContain('bg-white');
  });
});
