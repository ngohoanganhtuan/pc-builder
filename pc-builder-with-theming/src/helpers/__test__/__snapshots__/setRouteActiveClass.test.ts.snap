// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [setRouteActiveClass] helper Helper [setRouteActiveClass] default class name should match snapshot when active 1`] = `"nav-link text-dark d-flex flex-row align-items-center text-decoration-none gap-1 border rounded d-flex flex-row p-2 bg-warning border-0"`;

exports[`Test [setRouteActiveClass] helper Helper [setRouteActiveClass] default class name should match snapshot when not active 1`] = `"nav-link text-dark d-flex flex-row align-items-center text-decoration-none gap-1 border rounded d-flex flex-row p-2 bg-white"`;
